src/main/java/com/snowflake/kafka/connector/internal/FileNameUtils.java

```diff
@@ -58,23 +59,26 @@ static String fileName(String prefix, long start, long end)
    * @param table table name
    * @param partition partition id
    * @param offset record offset
+   * @param isKey is the broken record a key or a value
    * @return file name
    */
-  static String brokenRecordFileName(String appName, String table, int partition, long offset)
+  static String brokenRecordFileName(String appName, String table, int partition, long offset, boolean isKey)
```

Comment:
If key is broken or value is broken?

Reply:
Yes, currently if the record is broken, we only send value as raw bytes. Here I want to send both key and value as raw bytes to table stage, if either key or value is broken. This isKey will tell the generated filename to indicate whether the file contains key bytes or value bytes.